### PR TITLE
support env in babelrc

### PR DESCRIPTION
**Pull request by hexsprite**
This PR implements support for the env key in `.babelrc` files.

It looks first for `BABEL_ENV` then `NODE_ENV` then finally defaults to `development` per the Babel documented behavior.

Rationale: be able to add additional babel plugins/presets for testing environment for example (eg. `babel-plugin-istanbul`)

I wasn't sure where to add tests as I did not see any existing tests for the functionality.  Please let me know your feedback.

### CODE REVIEW

File: packages/babel-compiler/babel-compiler.js [hunk 1]
```diff
@@ -300,6 +300,16 @@ BCp._inferHelper = function (
   merge(babelOptions, babelrc, "presets");
   merge(babelOptions, babelrc, "plugins");
 
+  const babelEnv = (process.env.BABEL_ENV ||
+                    process.env_NODE_ENV ||
```
Comment (benjamn) [inline packages/babel-compiler/babel-compiler.js]: Should this be `process.env.NODE_ENV`?

**Comment by benjamn** [comment]
> I wasn't sure where to add tests as I did not see any existing tests for the functionality. Please let me know your feedback.

I would recommend adding `env` to some of the `.babelrc` files in [the modules test app](https://github.com/meteor/meteor/tree/devel/tools/tests/apps/modules), or creating new local packages with new `.babelrc` files in that app.

File: packages/babel-compiler/babel-compiler.js (hunk 1 above)
Comment (hexsprite) [inline packages/babel-compiler/babel-compiler.js]: I believe so. From the babel docs:

`The env key will be taken from process.env.BABEL_ENV, when this is not available then it uses process.env.NODE_ENV if even that is not available then it defaults to "development".`

https://babeljs.io/docs/usage/babelrc/#env-option

Comment (hwillson) [inline packages/babel-compiler/babel-compiler.js]: @hexsprite Ben's referring to the typo in your `process.env_NODE_ENV` reference (note the `_`).

Comment (hexsprite) [inline packages/babel-compiler/babel-compiler.js]: Oh, woops. Got it.

**Pull request merged**